datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnerModal.tsx

```diff
@@ -47,10 +76,96 @@ export const AddOwnerModal = ({ visible, onClose, refetch }: Props) => {
                 message.error({ content: `Failed to add owner: \n ${e.message || ''}`, duration: 3 });
             }
         }
+        setSelectedActor(undefined);
         refetch?.();
         onClose();
     };
 
+    // When a user search result is selected, set the urn as the selected urn.
+    const onSelectActor = (newUrn: string) => {
+        if (inputEl && inputEl.current) {
+            (inputEl.current as any).blur();
+        }
+        const filteredActors = combinedSearchResults
+            .filter((result) => result.entity.urn === newUrn)
+            .map((result) => result.entity);
+        if (filteredActors.length) {
+            const actor = filteredActors[0];
+            setSelectedActor({
+                displayName: entityRegistry.getDisplayName(actor.type, actor),
+                type: actor.type,
+                urn: actor.urn,
+            });
+        }
+    };
+
+    // When a user search result is selected, set the urn as the selected urn.
+    const onDeselectActor = (_: string) => {
+        setSelectedActor(undefined);
+    };
+
+    // Invokes the search API as the user types
+    const handleSearch = (type: EntityType, text: string, searchQuery: any) => {
+        console.log(text);
+        if (text.length > 2) {
+            searchQuery({
+                variables: {
+                    input: {
+                        type,
+                        query: text,
+                        start: 0,
+                        count: 5,
+                    },
+                },
+            });
+        }
```

Comment:
ah! so soon we will have `types: [type1, type2]` input & patterns like this will become simpler!

Reply:
right!